### PR TITLE
fix(util): fix run-as-root await rm bug

### PR DESCRIPTION
**Pull request by zetlen**
- Fix my mistake where I both ran `rm` in the shell and ran `unlink` in
 JS.
 - Properly await result of sudoPrompt.
 - Add tests against real FS to ensure that file cleanup is handled
 well.

Should fix magento-research/venia-pwa-concept#51 when merged and released.

### CODE REVIEW

File: src/util/run-as-root.js [hunk 1]
```diff
@@ -28,10 +28,13 @@ const sudoPromptToRunShell = async cmd =>
...
             debug(`sudo ${cmd} returned`, { error, stdout, stderr });
             if (error) {
-                error.message = [error.message, stderr, stdout]
-                    .filter(x => !!x)
-                    .join('\n\n');
-                return reject(error);
+                return reject(
+                    new Error(
+                        [error.message || error, stderr, stdout]
+                            .filter(x => !!x)
```
Comment (jimbo) [inline src/util/run-as-root.js]: Nitpick: can't this be simplified to `.filter(x => x)`?

File: src/util/__tests__/run-as-root.spec.js [hunk 2]
```diff
@@ -23,22 +25,53 @@ test('serializes and writes a script to fs', async () => {
...
 
 test('cleans up temp file on success or failure', async () => {
     sudoPrompt.exec.mockImplementationOnce((cmd, opts, cb) =>
-        setImmediate(() => cb(null, '', ''))
+        exec(cmd).then(res => cb(null, res.stdout, res.stderr), cb)
     );
-    await runAsRoot(() => true);
+    writeFile.mockImplementation(fs.writeFile);
+    unlink.mockImplementation(fs.unlink);
+    await runAsRoot(() => console.log('squanch'));
```
Comment (jimbo) [inline src/util/__tests__/run-as-root.spec.js]: What is this?

File: src/util/run-as-root.js (hunk 1 above)
Comment (DrewML) [reply src/util/run-as-root.js]: Can also be `.filter(Boolean)`

File: src/util/__tests__/run-as-root.spec.js [hunk 3]
```diff
@@ -4,6 +4,8 @@ jest.mock('sudo-prompt');
 const sudoPrompt = require('sudo-prompt');
 const path = require('path');
 const { writeFile, unlink } = require('../promisified/fs');
+const fs = require.requireActual('../promisified/fs');
```
[the remark this answers is not in the document]
Comment (zetlen) [reply src/util/__tests__/run-as-root.spec.js]: Yeah, I knew that looked ugly. Thank you for detailing the fix; I'll make that change.

File: src/util/__tests__/run-as-root.spec.js (hunk 2 above)
Comment (zetlen) [reply src/util/__tests__/run-as-root.spec.js]: I *knew* I wasn't using it right.

File: src/util/run-as-root.js [hunk 4]
```diff
@@ -28,10 +28,18 @@ const sudoPromptToRunShell = async cmd =>
         sudoPrompt.exec(cmd, opts, (error, stdout, stderr) => {
             debug(`sudo ${cmd} returned`, { error, stdout, stderr });
             if (error) {
-                error.message = [error.message, stderr, stdout]
-                    .filter(x => !!x)
+                const isTruthy = x => x;
```
Comment (DrewML) [inline src/util/run-as-root.js]: Nit: `isTruthy` >> `identity`. Name doesn't match implementation

**Comment by zetlen** [comment]
@jimbo this still awaits your review if you want to give it! Let me know if you want to abstain.
@DrewML Since your last requested change was so small, and I've made that change and pushed it...would you like to approve the PR?

**Pull request merged**